apps/ledger-live-mobile/src/screens/ReceiveFunds/03-Confirmation.tsx

```diff
@@ -111,12 +111,14 @@ export default function ReceiveConfirmation({ navigation, route }: Props) {
   useEffect(() => {
     if(route.params?.createTokenAccount && !hasAddedTokenAccount) {
       const newMainAccount = mainAccount;
-      const emptyTokenAccount = makeEmptyTokenAccount(newMainAccount, currency);
-      newMainAccount.subAccounts = [...newMainAccount.subAccounts, emptyTokenAccount];
+      if(!newMainAccount.subAccounts.find((acc: TokenAccount) => acc?.token?.id === currency.id)) {
+        const emptyTokenAccount = makeEmptyTokenAccount(newMainAccount, currency);
+        newMainAccount.subAccounts = [...newMainAccount.subAccounts, emptyTokenAccount];
 
-      dispatch(replaceAccounts({ scannedAccounts: [newMainAccount], selectedIds: [newMainAccount.id], renamings: {}}));
-      setIsToastDisplayed(true);
-      setHasAddedTokenAccount(true);
+        dispatch(replaceAccounts({ scannedAccounts: [newMainAccount], selectedIds: [newMainAccount.id], renamings: {}}));
```

Comment:
the use of replaceAccounts here isn't ideal. you must have a new action dedicated to this small addition you are doing it. otherwise, we risk some race condition, e.g. if a background synchronisation is happening and has updated an account, the replace action is going to override this.


Also, a bit before, we must not mutate `mainAccount ` in place, instead we must `newMainAccount = { ...mainAccount, subAccounts: `

mutating will have bad consequences and unexpected bugs as the reference is shared everywhere in the app. the rest of the code is good because you use the dispatch way, but only that part is problematic.

Reply:
i updated the mutation part :+1: 
for the new action we could create it later, i didn't have the time for it now